### PR TITLE
Migrate statistics to PouchDB - Closes #131

### LINKED ISSUE #131: Migrate transaction statistics to embedded database
### Description

Transaction statistics are currently backed by Postgres database. Migrate this feature to embedded database. Create an abstraction layer between the database API and Lisk Service blockchain data storage.

### Motivation

- Dropping Postgres database makes the further development easier and allows to maintain project without external dependencies
- Experience during this transition will be useful in the further development process

### Acceptance Criteria

- Transaction statistics are migrated to PouchDB
- Postgres dependencies are dropped
- There is an abstraction layer between PouchDB and blockchains storage that makes it database-agnostic

### PR DESCRIPTION
**Pull request by MichalTuleja**
### What was the problem?
This PR resolves #131

### How was it solved?
- [x] Add PouchDB to the project
- [x] Enable auto-compaction
- [x] Make statistics written into PouchDB

### How was it tested?

Jenkins

### CODE REVIEW

File: services/core/shared/transactionStatistics.js [hunk 1]
```diff
@@ -13,79 +13,122 @@
...
 			transactionStatisticsQueue.add({ date });
 		}
 	});
 };
 
-const init = async historyLengthDays => {
-	await ensureDbTableIsCreated();
-	fetchTransactionsForPastNDays(historyLengthDays);
-};
+const init = async historyLengthDays => fetchTransactionsForPastNDays(historyLengthDays);
```
Comment (nagdahimanshu) [inline services/core/shared/transactionStatistics.js:131]: is there any need to make it async?

File: services/core/shared/transactionStatistics.js [hunk 2]
```diff
@@ -13,79 +13,122 @@
...
-const database = getDatabase();
+	const result = await db.find({
+		selector: {
+			date: {
+				$gte: params.dateFrom.toISOString(),
+				$lte: params.dateTo.toISOString(),
+			},
+			// 	WHERE $<dateFrom> <= timestamp AND timestamp <= $<dateTo>
+		},
+	});
```
Comment (nagdahimanshu) [inline services/core/shared/transactionStatistics.js:33]: Isn't it better if we write a common function for this and reuse that function wherever needed in the file?

**Review by nagdahimanshu — CHANGES_REQUESTED** [review]

File: services/core/shared/transactionStatistics.js (hunk 1 above)
Comment (sameersubudhi) [reply services/core/shared/transactionStatistics.js:131]: `fetchTransactionsForPastNDays` being an async function, it is better to mark the `init` and `updateTodayStats` methods async.

File: services/core/shared/transactionStatistics.js (hunk 2 above)
Comment (sameersubudhi) [reply services/core/shared/transactionStatistics.js]: Done!

File: services/core/shared/transactionStatistics.js (hunk 1 above)
Comment (sameersubudhi) [reply services/core/shared/transactionStatistics.js]: `fetchTransactionsForPastNDays` is an async method. Added `await` keyword to make it explicit.

**Review by nagdahimanshu — APPROVED** [review]
Looks good!

**Comment by sameersubudhi** [comment]
> Looks good!

@MichalTuleja Kindly review the changes as well.

**Review by sameersubudhi — APPROVED** [review]

**Review by MichalTuleja — COMMENTED** [review]
Awesome work, thank you!

**Pull request merged**